### PR TITLE
feat: add a logo and some basics in the README

### PR DESCRIPTION
**Pull request by guseggert**

### CODE REVIEW

File: README.md
Comment (guseggert) [inline README.md]: this needs to be changed to the "main" branch before merging

Comment (guseggert) [inline README.md]:
```suggestion
<img src="https://raw.githubusercontent.com/ipfs/go-libipfs/main/logo.svg" alt="go-libipfs logo" title="go-libipfs logo" width="200">
```

File: README.md
Comment (hacdias) [inline README.md:2]: Maybe this is repeating too many bananas 😃

**Review by hacdias — APPROVED** [review]

**Comment by BigLep** [comment]
Thanks for flushing this out further Gus.  A couple of things I think would be great to add:
1. Link to docs.  I assume there are go docs generated from all the modules?  I'm thinking about how does someone get a sense of what tools are available to them in the go-libipfs toolbox?  Clicking into each directory is hit or miss with READMEs.  
2. Where someone should go to talk with the maintainers.  Is it just "open an issue" or are they welcome to join a channel?  (Maybe it makes sense for us to create #go-libipfs-maintainers if we don't want folks showing up to #ipfs-implementers)
3. Maybe have some expectations for each module that is added:
  - Clear README
  - codeowner file
  - code coverage badge

File: README.md
Comment (hacdias) [inline README.md:11]: A Go docs link:

```suggestion
[![Go Test](https://github.com/ipfs/go-libipfs/actions/workflows/go-test.yml/badge.svg)](https://github.com/ipfs/go-libipfs/actions/workflows/go-test.yml)
[![Go Docs](https://img.shields.io/badge/godoc-reference-blue.svg)](https://pkg.go.dev/github.com/ipfs/go-libipfs)
```

File: README.md
Comment (guseggert) [reply README.md:2]: I wanted to leave a copy-pastable unicode banana there too, the big one is not copy-pastable (and github won't let me make text that big).  This was also inspired by Downshift's README https://github.com/downshift-js/downshift/blob/master/README.md

File: README.md
Comment (guseggert) [reply README.md:11]: ah nice, added

**Comment by guseggert** [comment]
> Thanks for flushing this out further Gus. A couple of things I think would be great to add:
> 
>     1. Link to docs.  I assume there are go docs generated from all the modules?  I'm thinking about how does someone get a sense of what tools are available to them in the go-libipfs toolbox?  Clicking into each directory is hit or miss with READMEs.
> 
>     2. Where someone should go to talk with the maintainers.  Is it just "open an issue" or are they welcome to join a channel?  (Maybe it makes sense for us to create #go-libipfs-maintainers if we don't want folks showing up to #ipfs-implementers)
> 
>     3. Maybe have some expectations for each module that is added:
> 
> 
>     * Clear README
> 
>     * codeowner file
> 
>     * code coverage badge

1) I've added a godoc link to https://pkg.go.dev/github.com/ipfs/go-libipfs (the auto-generated Go docs, this is pretty standard for Go libs).

2) Good idea, the assumption is "open an issue" but Slack's a good comm channel too.

3) Also good ideas, will add.

File: README.md
Comment (BigLep) [inline README.md:62]: Fine for now, but we can also add IPFS Discord and ipfs.io Matrix to make it easier for people to meet us where they are.

File: README.md
Comment (BigLep) [inline README.md:13]: Sweet.  Do we have any CI checks to make sure we don't regress (at least that we don't regress when making non-import code changes)?

**Review by BigLep — APPROVED** [review]
Good improvements - thanks Gus!

File: README.md
Comment (guseggert) [reply README.md:13]: Yes I will add them in a subsequent PR

**Pull request merged**
